### PR TITLE
gh-474 Footer should include release specific links

### PR DESCRIPTION
**Pull request by ghillert**
* Change the way we retrieve AboutInfo to prevent multiple HTTP requests
  - Use a BehaviorSubject to propagate changes to the various subscribers
* Create dedicated domain objects modeled after the `AboutResource` classes from the server
* Create tests for each domain object
* Fix and polish tests

resolves https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/474

### CODE REVIEW

**Comment by cppwfs** [comment]
Rebased against master and ran against this branch.  However when the dashboard comes up the busy pops up and stays.

**Comment by cppwfs** [comment]
Spinner does not go away for any tabs.

**Comment by cppwfs** [comment]
Moving from one tab to another is leaving artifacts from the previous tab

File: ui/src/app/shared/model/about/dependency.model.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { FeatureInfo } from './feature-info.model';
```
Comment (cppwfs) [inline ui/src/app/shared/model/about/dependency.model.spec.ts]: can remove FeatureInfo import.

File: ui/src/app/shared/model/about/runtime-environment-details.model.spec.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { FeatureInfo } from './feature-info.model';
```
Comment (cppwfs) [inline ui/src/app/shared/model/about/runtime-environment-details.model.spec.ts]: Don't need FeatureInfo & Dependency imports

File: ui/src/app/shared/model/about/runtime-environment-details.model.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { Serializable } from '../serialization/serializable.model';
+import { Dependency } from './dependency.model';
```
Comment (cppwfs) [inline ui/src/app/shared/model/about/runtime-environment-details.model.ts]: Can remove Dependency import

**Comment by ghillert** [comment]
@cppwfs Addressed the issues

**Comment by cppwfs** [comment]
Do we want to retire key-value-filter since it doesn't seem to be used anymore.  If so, do we want to open another PR to do that.  Thoughts?

File: ui/src/app/about/about.component.spec.ts [hunk 4]
```diff
@@ -68,17 +68,17 @@ describe('AboutComponent', () => {
     expect(des[11].nativeElement.textContent).toContain('https://github.com/spring-cloud/spring-cloud-dataflow/issues');
   });
 
-  it('Should display headers only on no server available.', () => {
-    aboutService.isAboutInfoAvailable = false;
-    fixture.detectChanges();
-    let de = fixture.debugElement.query(By.css('h1'));
-    let el = de.nativeElement;
-    expect(el.textContent).toContain('About');
+  // fit('Should display headers only on no server available.', () => {
```
[the remark this answers is not in the document]
Comment (cppwfs) [reply ui/src/app/about/about.component.spec.ts]: Thoughts on this code segment?

File: ui/src/app/shared/model/about/about-info.model.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { FeatureInfo } from './feature-info.model';
+import { VersionInfo } from './version-info.model';
+import { SecurityInfo } from './security-info.model';
+import { RuntimeEnvironment } from './runtime-environment.model';
+import { Serializable } from '../serialization/serializable.model';
+
+export class AboutInfo implements Serializable<AboutInfo> {
```
Comment (cppwfs) [inline ui/src/app/shared/model/about/about-info.model.ts]: Add typescript doc.  Take credit for the great work!

File: ui/src/app/shared/services/shared-about.service.ts [hunk 6]
```diff
@@ -11,42 +11,52 @@ import 'rxjs/add/observable/of';
 import { ErrorHandler } from '../model';
 import { FeatureInfo } from '../model/about/feature-info.model';
 import { Subject } from 'rxjs/Subject';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { AboutInfo } from '../model/about/about-info.model';
 
 @Injectable()
 export class SharedAboutService {
```
Comment (cppwfs) [inline ui/src/app/shared/services/shared-about.service.ts]: Probably want to add some typescript doc here as well.

File: ui/src/app/shared/services/shared-about.service.ts [hunk 7]
```diff
@@ -11,42 +11,52 @@ import 'rxjs/add/observable/of';
...
 
-  public extractData(res: Response) {
-    console.log('extract data', res);
-    const body = res.json();
-    this.aboutInfo = body;
-    this.featureInfo = new FeatureInfo().deserialize(body.featureInfo);
-    this.featureInfoSubject.next(this.featureInfo);
-    return body;
+  public extractData(response: Response): AboutInfo {
+    const aboutInfo = new AboutInfo().deserialize(response.json());
```
Comment (cppwfs) [inline ui/src/app/shared/services/shared-about.service.ts]: Why not assign this directly to the this.aboutInfo

**Comment by cppwfs** [comment]
Ran additional tests on it.  In unsecured mode it does fine.  However with security enabled only the `Apps` and `About` tabs appear.  Just to make sure it wasn't brought in from the last rebase I went back to master and tested security there and it was ok.

**Comment by ghillert** [comment]
@cppwfs Issue with security should be addressed now.

**Comment by ghillert** [comment]
@cppwfs Rebased and squashed commits

**Comment by cppwfs** [comment]
LGTM Some small lint changes.   I'll fix them on merge.

**Comment by cppwfs** [comment]
Fixed Lint issues.  Rebased, Squashed Merged.

**Pull request closed without merging**